stats/team.py

```diff
@@ -56,3 +55,21 @@ def populate(self):
         Bulk insert teams.
         """
         Team.insert_many(self.rows).execute()
+
+    def get_required_fields(self):
+        """
+        Get list of required fields to pull from the result set headers.
+        """
+        return [
```

Comment:
Why are there more fields here than what you are using in the new_row construction loop?

So, imo, it's best to consolidate this information in one place. That place already exists - the model. Keeping it in two places means that they need to be in sync every time we want to add a new column.

I was looking at how this information can be extracted from the model. There are a couple of options - 

1. From the model directly

You can import the model into the requester, and dig through its internals to get the columns described in the model.

```
In [10]: Team._meta.columns
Out[10]:
{'team_id': <IntegerField: Team.team_id>,
 'abbreviation': <CharField: Team.abbreviation>,
 'nickname': <CharField: Team.nickname>,
 'year_founded': <CharField: Team.year_founded>,
 'city': <CharField: Team.city>}

In [11]: Team._meta.fields
Out[11]:
{'team_id': <IntegerField: Team.team_id>,
 'abbreviation': <CharField: Team.abbreviation>,
 'nickname': <CharField: Team.nickname>,
 'year_founded': <CharField: Team.year_founded>,
 'city': <CharField: Team.city>}

In [12]: Team._meta.sorted_field_names
Out[12]: ['team_id', 'abbreviation', 'nickname', 'year_founded', 'city']
```

This is easy, but the only reason I am not super onboard this is because we will be accessing a private object `_meta`. Python doesn't have strong object access controls, and it's a convention to try and avoid accessing a private member or methods of a private member because the owner could change the behavior at any time.

2. From db object.

Since the settings object is being passed to every requester, it's possible to get this info from the db object. 

```
In [28]: db.get_columns('team')
Out[28]:
[ColumnMetadata(name='team_id', data_type='integer', null=False, primary_key=True, table='team', default=None),
 ColumnMetadata(name='abbreviation', data_type='character varying', null=True, primary_key=False, table='team', default=None),
 ColumnMetadata(name='nickname', data_type='character varying', null=True, primary_key=False, table='team', default=None),
 ColumnMetadata(name='year_founded', data_type='character varying', null=True, primary_key=False, table='team', default=None),
 ColumnMetadata(name='city', data_type='character varying', null=True, primary_key=False, table='team', default=None)]

In [29]: columns = db.get_columns('team')

In [30]: columns[0].name
Out[30]: 'team_id'
```

I think this is clean.

>  but I don't think 100% of the DB column names map to the rowset names

Going with the above approach, since the column names returned in the response are all caps, we can switch between lower upper as we query info from the model.

What do you think?

Reply:
I agree it is cleaner to pull from the metadata, but there is an edge case: `player_game_log`, `player_season` and `player_gtt` all use the `season_id` that is passed to the fetch/build function, and not from the response. So handling non-existent but still valid keys will be needed. Maybe that isn't so bad, though. I'll take your suggestion out for a spin and ping you for another review.